### PR TITLE
refactor(python): Accept optional extra packages to install

### PR DESCRIPTION
**Pull request by mbudnek**
I'm planning to add the keyrings.envvars package to be installed with the pip-tools tool, so instead of copy/pasting the common python install_tool logic a second time this refactors it into a common function that will install any packages given as arguments along with the tool being installed.

### CODE REVIEW

File: src/usr/local/containerbase/utils/python.sh
Comment (viceice) [inline src/usr/local/containerbase/utils/python.sh:33]:
```suggestion
# shellcheck disable=SC2120
```
I think we use the prefix on other places

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — APPROVED** [review]
next time use the re-request review button next to the pr reviewers 😉

**Comment by viceice** [comment]
please don't merge main, branch doesn't have to be up to date. we use merge queue which will do a squash merge and test it before merging to target.

**Review by viceice — APPROVED** [review]

**Comment by mbudnek** [comment]
@viceice I merged main just to get the tests to re-run, since some sort of docker daemon issue caused them to fail.

**Comment by viceice** [comment]
ok, ping me next time. I can simply rerun only failed jobs, which is faster than a full run

**Pull request merged**
